src/components/Dashboard/Main.js

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import React, { Component, Fragment } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  Flavors,
+  DashboardHome as Home,
+  Migrations,
+  DashboardNotFound as NotFound,
+  Roles,
+  RoleAdd,
+  RoleEdit,
+  RoleDelete,
+  RoleUsers,
+  RoleAddUser,
+  RoleDeleteUser,
+  Users,
+  UserRoles
+} from 'components/Dashboard/';
+
+import { actions as dashboardActions } from 'reducers/dashboard';
+import { getDashboardComponent } from 'selectors/dashboard';
+
+class DashboardMain extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    dashboardComponent: PropTypes.object.isRequired
+  };
+  display() {
+    const { dashboardComponent } = this.props;
+    const { name, item } = dashboardComponent;
+    const defaultOpts = {
+      border: false,
+      header: true,
+      title: false,
+      style: {}
+    };
+
+    switch (name) {
```

Comment:
You could export a const (effectively an enum) with the component names and then create a map like

```jsx
export const ComponentNames = {
  HOME: 'Home'
};
export const Components = {
  ComponentNames.HOME: <Home />,
};
```

Then to render

```jsx
render() {
  const { dashboardComponent } = this.props;
  const { name, item } = dashboardComponent;
  const Component = Components[name];

  return <Component {...this.props} />;
}
```

I don't mind the switch but this might be a case where having the enum is useful. No need to change it unless you want to.